pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,726 @@
+                );
+
+                // get the highest bidder
+                let highest_bidder = auction.bidders.get_highest_bid().cloned();
+
+                // Transfer nft to highest bidder
+                match highest_bidder {
+                    Some(bidder) => {
+                        // remove highest bidder from list to avoid claim
+                        let bidder = auction
+                            .bidders
+                            .remove_bid(bidder.0)
+                            .take()
+                            .ok_or(Error::<T>::UnexpectedError)?;
+
+                        // Handle marketplace fees
+                        let marketplace_info =
+                            T::MarketplaceHandler::get_marketplace_info(auction.marketplace_id)
+                                .ok_or(Error::<T>::UnexpectedError)?;
+                        let marketplace_commission_amount =
+                            bidder.1.saturating_mul(marketplace_info.1.into()) / 100u32.into();
+                        let amount_to_auction_creator = bidder
+                            .1
+                            .checked_sub(&marketplace_commission_amount)
+                            .ok_or(Error::<T>::UnexpectedError)?;
+
+                        // Transfer fee to marketplace
+                        T::Currency::transfer(
+                            &Self::account_id(),
+                            &marketplace_info.0,
+                            marketplace_commission_amount,
+                            KeepAlive,
+                        )?;
+
+                        // Transfer remaining to auction creator
+                        T::Currency::transfer(
+                            &Self::account_id(),
+                            &auction.creator,
+                            amount_to_auction_creator,
+                            KeepAlive,
+                        )?;
+
+                        // transfer NFT to highest bidder
+                        T::NFTHandler::set_owner(nft_id, &bidder.0)?;
+                        // Mark NFT as not listed for sale
+                        T::NFTHandler::set_listed_for_sale(nft_id, false)?;
+                    }
+                    None => (),
+                }
+
+                // ensure the state is completed so other users can claim amount
+                auction.state = AuctionState::Completed;
+
+                // emit bid created event
+                Self::deposit_event(Event::AuctionCompleted {
+                    nft_id,
+                    marketplace_id: auction.marketplace_id,
+                    creator: auction.creator.clone(),
+                });
+
+                Ok(())
+            })?;
+
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        #[transactional]
+        pub fn claim_bid(origin: OriginFor<T>, nft_id: NFTId) -> DispatchResultWithPostInfo {
```

Comment:
In the next iteration, we need to separate claims out of the auction. Once the auction has ended, we delete it and if there are any claims left, we move them into a new separate structure.